API set up

darthrighteous opened the pull request:
### **What does this PR do?**
Have CRUD API endpoints set up.

### **Description of Tasks to be completed**
Have the following endpoints, validation logic and tests for them working:

1. `GET /v1/questions?sortBy=score` // retrieves the list of questions (optional score sort query)
2. `GET /v1/questions/:questionId` // retrieves a specific question
3. `POST /v1/questions` // posts a question
4. `POST /v1/questions/:questionId/answers` // posts an answer to a question
5. `PATCH /v1/questions/:questionId/answers/:answerId` // accepts an answer
6. `PATCH /v1/questions/:questionId` // upvotes/downvotes or edits a question's details
7. `DELETE /v1/questions/:questionId` // deletes a quesition

> App hosted on heroku at https://fast-river-42221.herokuapp.com/

### **How should this be manually tested**
- Clone the repository, cd into it, checkout feature branch
- Run the following `npm run build` `npm run serve` `npm run start`
- Use postman to test all end points using http://localhost:4001/ (e.g for endpoint 2 use http://localhost:4001/v1/questions/1)

 **some endpoints require request body input**
- endpoint 3/4 - JSON request body 

> example

```
{
	"title": "dummy title",
	"body": "dummy body",
	"date": "dummy date",
	"user": "darth"
}
```
- endpoint 6 - JSON request body  (**attribute** can be '_score_', _'title'_, _'body'_)
    - if **attribute** is _score_, then _value_ can be '_increment_' or '_decrement_' to upvote/downvote respectively.
    - if **attribute** is title or body, value should be whatever new string to replace.

> example

```
{
	"attribute": "score",
	"value": "increment"
}
```
### **What are relevant pivotal tracker stories?**
#159784977 #159784908 #159784912 #159784920 #159784926 #159784945 #159784945

Billmike commented on utils.js:
```diff
@@ -0,0 +1,113 @@
+  body: 'How do you divide a prime number by itself?',
+  date: '20/03/2003',
+  user: 'tyronne',
+};
```
New line after each object declaration.

Darthrighteous replied:
done

Billmike commented:
1. Fix the merge conflicts you have.
2. 87 commits?? 😱

Darthrighteous commented on .babelrc:
```diff
@@ -0,0 +1 @@
+{ "presets": ["env"] }
```
done

Darthrighteous commented on .coveralls.yml:
```diff
@@ -0,0 +1 @@
+repo_token: qjtcWieTaMQEznKQOpRAIkz6xSaBFeq2Y
```
done

Darthrighteous commented on .eslintrc:
```diff
@@ -0,0 +1,3 @@
+{
+	"extends": "airbnb-base"
+}
```
done

Darthrighteous commented on .gitignore:
```diff
@@ -0,0 +1 @@
+/node_modules
```
done

Darthrighteous commented on .travis.yml:
```diff
@@ -0,0 +1,9 @@
+    - "node_modules"
+
+after_success:
+  - npm run coverage
```
done

Darthrighteous commented:
@Billmike Thanks for the review. I think I've made all the corrections pointed out. 
Please let me know if I've miss anything!

Darthrighteous commented:
Added heroku link

Darthrighteous commented:
I have implemented es-lint rules for valid-jsdoc and max-len.

darthrighteous committed:
feat: create API endpoints and tests

    -create CRUD endpoints and tests for the following http verbs - GET, POST, PATCH, DELETE to achieve the following:
        -get all questions in question array
        -get a specific question by id
        -post a question in question array
        -post an answer to a specific question
        -accept an answer to a specific question
        -upvote/downvote a question
        -edit a questions details (body/title)
        -delete a question
    -integrate travis CI, coveralls
    - apply airbnb style guide
    - change common js export and require statements to match ES6 standard named/default export and import statements
    - add babel functionality to transpile down
    - coverage reporter from istanbul to nyc for better support
    - remove UI folder from branch
    - host app on heroku at https://fast-river-42221.herokuapp.com/

[(finishes) #159784945]

darthrighteous committed:
update README
